client/state/login/actions.js

```diff
@@ -572,3 +575,43 @@ export const logoutUser = redirectTo => ( dispatch, getState ) => {
 			return Promise.reject( error );
 		} );
 };
+
+/**
+ * Retrieves the type of authentication of the account of the specified user.
+ *
+ * @param {String} usernameOrEmail - id of the user
+ * @return {Function} a promise that will resolve once the authentication account type has been retrieved
+ */
+export const getAuthAccountType = usernameOrEmail => dispatch => {
+	dispatch( {
+		type: LOGIN_AUTH_ACCOUNT_TYPE_REQUEST,
+	} );
+
+	if ( usernameOrEmail === '' ) {
+		return new Promise( ( resolve, reject ) => {
+			const error = {
+				code: 'empty_username',
+				message: translate( 'Please enter a username or email address.' ),
+				field: 'usernameOrEmail',
+			};
+
+			defer( () => {
```

Comment:
Why do we need to reject to promise asynchronously here?

Reply:
Because of an issue mentioned on Slack in p1510498319000092-slack-amber-dev, discussed in p1510504100000034-slack-amber-dev, and fixed by Yury in https://github.com/Automattic/wp-calypso/pull/19688/commits/b93bb3018169ed0a84f8062b7c8dc66141a779c2. Basically the `Email Address or Username` field is not focused the second time your click on the `Continue` button if this field is empty.